moved custom_spec['ipsettings'] section inside check for key

jberry opened the pull request:
fixes #41

plribeiro3000 commented on lib/fog/vsphere/requests/compute/vm_clone.rb:
```diff
@@ -268,21 +268,21 @@ def vm_clone(options = {})
                 custom_spec['globalIPSettings']['dnsSuffixList'] = custom_spec['dnsSuffixList'] || [custom_spec['domain']] if ( custom_spec['dnsSuffixList'] || custom_spec['domain'])
+                if (custom_spec['ipsettings'].key?('ip') or custom_spec['ipsettings'].key?('gateway') or custom_spec['ipsettings'].key?('subnetMask') or custom_spec['ipsettings'].key?('domain') or custom_spec['ipsettings'].key?('dnsServerList'))
+                    if custom_spec['ipsettings'].key?('ip') 
+                      raise ArgumentError, "subnetMask is required for static ip" unless custom_spec["ipsettings"].key?("subnetMask")
```
Maybe you can combine this inline unless with the parent if conditional to improve readability?

jrgarcia commented on lib/fog/vsphere/requests/compute/vm_clone.rb:
```diff
@@ -268,21 +268,21 @@ def vm_clone(options = {})
+                    custom_spec['nicSettingMap'][0]['adapter']['dnsServerList'] = custom_spec['ipsettings']['dnsServerList'] if custom_spec['ipsettings'].key?('dnsServerList')
+                 end       
               end
         
```
I'd also remove this empty line. Not a big deal, but it helps with following the nesting.

jberry commented:
Moved inline unless into parent if statement for easier readability per plribeiro3000
Removed empty line per jrgarcia
Fixed end alignment to please codeclimate

plribeiro3000 commented:
Code wise LGTM. @jrgarcia have more insights on the platform to move this forward!

jrgarcia commented:
@plribeiro3000 Travis is failing the 1.9.x builds. It's because of dependencies though. I'm installing 1.9.3 on my machine right now. If just pinning those dependencies fixes things, I'll merge this in and then update the gemspec with those dependencies pinned and release a new version of fog-vsphere. @plribeiro3000 What do you think about making a 1.0.0 release soon?

jberry commented:
For what it's worth, I didn't add or remove any existing code (with the exception of that change of the inline unless into the if statement).  

All I did was move an existing if block inside another existing if block to prevent the nil case

jrgarcia commented:
Well, everything seems to run just fine locally with 1.9.3 and JRuby in 1.9 mode.

plribeiro3000 commented:
@jrgarcia Go ahead with the `1.0.0` release. 👍
